plugins/json/README.md

```diff
@@ -0,0 +1,197 @@
+
+The JSON plugin doesn't yet support the function of waiting for multiple variables to be readable before writing them.
+
+## Configuration grammar {#json_configuration_grammar}
+
+There are two syntax for the JSON configuration file:
+First one is `var_name: file_path.json`, as for example :
+
+```yaml
+plugins:
+  json:
+    data1: file_path.json 
+```
+
+The second one starts with `- file: file_path.json` with arguments below
+
+|key|value|
+|:--|:----|
+|`"file"` (*mandatory*)| Name of the output file|
+|`"write"` (*mandatory*)| The variables to be writen|
+|`"when"` (*optional*)| Indicate some kind of condition|
+
+For example,
+
+```yaml
+plugins:
+  json:
+    - file: file_path.json
+      when: iteration % 10 = 0 # This is optional
+      write: [data1, data2, ...]
+```
+
+Note : by default, if not specified a true condition is set, so that the variable is written every time PDI is granted read permission.
+
+## Test the JSON plugin ! {#json_test_plugin}
+
+The easiest way to test the plugin is refering to the example section, that provides a full somewhat realistic C simulation code, where PDI is enabled and the JSON plugin is used. This one can be found at \ref PDI_example
+
+To use it, first compile using CMake with `BUILD_TESTING` set to `ON`, then execute : 
+```bash
+build/PDI_EXAMPLE/PDI_example_C example/json.yml
+```
+
+<!-- ### Example output -->
+
+## Full configuration file example {#json_config_example}
+
+\snippet "../../example/json.yml" example_config_json_plugin
+
+## On going work / Limitations {#json_on_going_work}
+
+- [ ] Adding support for reading variables. 
+    Currently, the JSON plugin doesn't support reading variables, only writing them. To cover more use cases, the JSON plugin should be able to read variables as well as write them.
+- [x] Improve strings handling. Concatenate arrays of chars to have "var": "value" instead of "var": ["v", "a", ...].
+- [x] Added full support for types that can be recursive, e.g. arrays of arrays, arrays within structures etc. 
+- [ ] The ability to create an on-the-go structures for several variables to be produced in a single JSON object
+- [x] Continuous integration: More test units coverage
+
+## The JSON plugin for custom formating {#json_custom_format}
+
+Initially, the aim of this plugin was to enable the writing of customized file formats. To achieve this, the template systems are logical and first-hand, in particular `mustache`. The latter, however, allows you to take JSON as input and format it.
+
+For the sake of the example, let's imagine the following code, creating which performs some kind of heat transfer simulation.
+
+### The PDI exposes
+
+<!-- \image html json_simulation_code_light.png width=800px -->
+
+```C
+    PDI_multi_expose("init",
+        "simulation_name", &cfg.simulation_name, PDI_OUT,
+        "max_steps", &cfg.max_steps, PDI_OUT,
+        "mesh_config", &cfg.mesh_config, PDI_OUT,
+        "rank", &rank, PDI_OUT,
+        NULL) ;
```

Comment:
I suggest to add a small definition of what is cfg.

Reply:
I removed the cfg prefix